pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/LinguisticNamingRule.java

```diff
@@ -132,10 +140,12 @@ private void checkGetters(ASTMethodDeclaration node, Object data, String nameOfM
     }
 
     private void checkSetters(ASTMethodDeclaration node, Object data, String nameOfMethod) {
-        ASTResultType resultType = node.getResultType();
-        if (hasPrefix(nameOfMethod, "set") && !resultType.isVoid()) {
-            addViolationWithMessage(data, node, "Linguistics Antipattern - The setter ''{0}'' should not return any type except void linguistically",
-                    new Object[] { nameOfMethod });
+        if (!hasIgnoredAnnotation(node)) {
```

Comment:
I'd move this to line 83 (within `visit(ASTMethodDeclaration node, Object data)`), so we ignore all overridden methods and not just setters